osquery/tables/system/darwin/keychain_utils.cpp

```diff
@@ -72,151 +73,132 @@ std::string getKeychainPath(const SecKeychainItemRef& item) {
-  for (CFIndex i = 0; i < CFArrayGetCount((CFArrayRef)alg); i++) {
-    alg_item = (CFDictionaryRef)CFArrayGetValueAtIndex((CFArrayRef)alg, i);
-    label = (CFStringRef)CFDictionaryGetValue(alg_item, kSecPropertyKeyLabel);
-    value = (CFStringRef)CFDictionaryGetValue(alg_item, kSecPropertyKeyValue);
-
-    if (expected_label.compare(stringFromCFString(label)) == 0) {
-      return stringFromCFString(value);
-    }
+  auto nid = OBJ_txt2nid("CN");
+  auto index = X509_NAME_get_index_by_NID(subject_name, nid, -1);
+  if (index == -1) {
+    return "";
   }
 
-  // Unknown algorithm OID.
-  return "";
-}
-
-std::string genSHA1ForCertificate(const SecCertificateRef& ca) {
-  CFDataRef ca_data;
+  auto commonNameEntry = X509_NAME_get_entry(subject_name, index);
+  if (commonNameEntry == nullptr) {
+    return "";
+  }
 
-  // Access raw data, hash and release.
-  ca_data = SecCertificateCopyData(ca);
-  auto digest = hashFromBuffer(
-      HASH_TYPE_SHA1, CFDataGetBytePtr(ca_data), CFDataGetLength(ca_data));
-  CFRelease(ca_data);
-  return digest;
+  auto commonNameData = X509_NAME_ENTRY_get_data(commonNameEntry);
+  auto data = ASN1_STRING_data(commonNameData);
+  return std::string(reinterpret_cast<char*>(data));
 }
 
-CFNumberRef CFNumberCreateCopy(const CFNumberRef& number) {
-  // Easy way to get allow releasing numbers existing in arrays/dicts.
-  // This follows Apple's guidance for "Create" APIs, caller controls memory.
-  CFNumberRef copy;
-  unsigned int value;
+std::string genHumanReadableDateTime(ASN1_TIME* time) {
+  BIO* bio_stream = BIO_new(BIO_s_mem());
+  if (bio_stream == nullptr) {
+    return "";
+  }
 
-  if (!CFNumberGetValue(number, kCFNumberIntType, &value)) {
-    return nullptr;
+  // ANS1_TIME_print's format is: Mon DD HH:MM:SS YYYY GMT
+  // e.g. Jan 1 00:00:00 1970 GMT (always GMT)
+  auto buffer_size = 32;
+  char buffer[32] = {0};
+  if (!ASN1_TIME_print(bio_stream, time)) {
+    BIO_free(bio_stream);
+    return "";
   }
 
-  copy = CFNumberCreate(nullptr, kCFNumberIntType, &value);
-  return copy;
+  // BIO_gets() returns amount of data successfully read or written
+  // (if the return value is positive) or that no data was successfully
+  // read or written if the result is 0 or -1.
+  if (BIO_gets(bio_stream, buffer, buffer_size) <= 0) {
+    BIO_free(bio_stream);
+    return "";
+  }
+  BIO_free(bio_stream);
+  return std::string(buffer);
 }
 
-CFDataRef CreatePropertyFromCertificate(const SecCertificateRef& cert,
-                                        const CFTypeRef& oid) {
-  // Set the list of attributes.
-  auto keys = CFArrayCreateMutable(nullptr, 0, &kCFTypeArrayCallBacks);
-  CFArrayAppendValue(keys, oid); // SecCertificateOIDs.h
-
-  // Request dictionary of dictionaries (one for each attribute).
-  auto certificate_values = SecCertificateCopyValues(cert, keys, nullptr);
-  CFRelease(keys);
-  if (certificate_values == nullptr) {
-    return nullptr;
+time_t genEpoch(ASN1_TIME* time) {
```

Comment:
I was really really surprised that openssl didn't handle timestamps as unix time, but have their own string representation. To make matters worse, I wasn't able to find any good docs.